pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -819,11 +819,42 @@ func (r *ReconcileClusterDeployment) setAdminKubeconfigStatus(cd *hivev1.Cluster
 	return nil
 }
 
+// ensureManagedDNSZoneDeleted is a safety check to ensure that the child managed DNSZone
+// linked to the parent cluster deployment gets a deletionTimestamp when the parent is deleted.
+// Normally we expect Kube garbage collection to do this for us, but in rare cases we've seen it
+// not working as intended.
+func (r *ReconcileClusterDeployment) ensureManagedDNSZoneDeleted(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) error {
+	if cd.Spec.ManageDNS {
+		dnsZone := &hivev1.DNSZone{}
+		dnsZoneNSN := types.NamespacedName{Namespace: cd.Namespace, Name: dnsZoneName(cd.Name)}
```

Comment:
style: use consistent name for the variable `dnsZoneNamespacedName` is used below

Reply:
Updates for all of these pushed.